### PR TITLE
Update amp-form docs to match validators and runtime updates.

### PR DESCRIPTION
**Pull request by mkhatib**

### CODE REVIEW

File: extensions/amp-form/amp-form.md [hunk 1]
```diff
@@ -71,19 +71,21 @@ __required__
...
 Action must be provided, `https` and is non-cdn link (does **NOT** link to https://cdn.ampproject.org).
 
 __Note__: `target` and `action` will only be used for non-xhr GET requests. AMP runtime will use `action-xhr` to make the request and will ignore `action` and `target`. When `action-xhr` is not provided AMP would make a GET request to `action` endpoint and use `target` to open a new window (if `_blank`). AMP runtime might also fallback to using action and target in cases where `amp-form` extension fails to load.
 
 **action-xhr**
-__(optional)__ for `GET` __required__ for `POST` requests 
+__required__ when `method=POST`
+__optional__ when `method=GET`
+
 You can also provide an action-xhr attribute, if provided, the form will be submitted in an XHR fashion.
```
Comment (cramforce) [inline extensions/amp-form/amp-form.md]: We should at least once explain what an XHR request is

File: extensions/amp-form/0.1/amp-form.js [hunk 2]
```diff
@@ -211,9 +211,12 @@ export class AmpForm {
         this.renderTemplate_(error.responseJson || {});
         rethrowAsync('Form submission failed:', error);
       });
-    } else if (this.target_ == '_top' && this.method_ == 'POST') {
-      this.cleanupRenderedTemplate_();
-      this.setState_(FormState_.SUBMITTING);
+    } else if (this.method_ == 'POST') {
+      e.preventDefault();
+      user().assert(false,
```
Comment (cramforce) [inline extensions/amp-form/0.1/amp-form.js]: Please add a test that throws this exception

Comment (mkhatib) [inline extensions/amp-form/0.1/amp-form.js]: Done

File: extensions/amp-form/amp-form.md (hunk 1 above)
Comment (mkhatib) [inline extensions/amp-form/amp-form.md]: Done

**Comment by mkhatib** [comment]
PTAL 👀

File: extensions/amp-form/amp-form.md [hunk 3]
```diff
@@ -71,18 +71,22 @@ __required__
...
 
-This attribute can be the same or a different endpoint than `action` and has the same action requirements above.
+An XHR request, is where the browser would make the request without a full load of the page or opening a new page. Browsers will send the request in the background using [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) when available and fallback to [XMLHttpRequest API](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest) for older browsers.
```
Comment (cramforce) [inline extensions/amp-form/amp-form.md]: No `,` after request. End first sentence in 'also sometimes called Ajax request.'

Comment (mkhatib) [inline extensions/amp-form/amp-form.md]: Done

**Pull request merged**